plugins/ril.c

```diff
@@ -284,7 +281,9 @@ static void ril_set_online_cb(struct ril_msg *message, gpointer user_data)
 	struct ril_data *rd = cbd->user;
 	ofono_modem_online_cb_t cb = cbd->cb;
 
-	if (message->error == RIL_E_SUCCESS) {
+	if (message != NULL && message->error == RIL_E_SUCCESS) {
```

Comment:
The only reason you need to check for message not being NULL is due to fact that you're now invoking func() directly in ril_send_power() with message == NULL if a ril_send fails.  In all other cases, the gril core guarantees that message is always valid.